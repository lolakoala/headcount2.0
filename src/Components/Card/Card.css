.card {
  color: #334855;
  display: flex;
  flex-direction: column;
  align-items: center;
  border: 2px solid #555555;
  border-radius: 5%;
  box-shadow: 4px 4px 2px #555555;
  width: 18%;
  margin: 5px 5px 5px 5px;
  background-color: #ACD0DD; }

.card:hover {
  background-color: #A2BBC2; }

.card-title {
  font-size: 16px;
  font-weight: bold; }

.aboveHalf {
  font-weight: bold; }

.belowHalf {
  font-weight: lighter;
  color: #7D215D; }
