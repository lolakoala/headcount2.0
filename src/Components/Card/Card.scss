@import '../css/variables.scss';

ul {
  list-style: none;
}

.card {
  color: #334855;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  border: 2px solid #555555;
  border-radius: 5%;
  box-shadow: 4px 4px 2px #555555;
  width: 12%;
  margin: 20px 20px 20px 20px;
  background-color: #ACD0DD;
}

.card:hover{
  cursor: pointer;
}

.hover:hover {
  .hover{background-color: $card-hover-color;}
  background-color: $card-hover-color;
  cursor: pointer;
}


.card-title {
  font-size: 16px;
  font-weight: bold;
}


.aboveHalf {
  font-weight: bold;

}

.belowHalf {
  font-weight: lighter;
  color: $poop;
}

.clear-button {
  width: 300px;
  height: 70px;
  border-radius: 25px;
  background-color: black;
  border: none;
  color: white;
  font-size: 24px;
  font-weight: bold;
}

.compare-and-button {

}
