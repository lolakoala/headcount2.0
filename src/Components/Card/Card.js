import React from 'react';
import './Card.css';
import PropTypes from 'prop-types';
import checkMark from '../../../assets/checkmark-xl.png';
import xMark from '../../../assets/x-mark-xl.png';

const Card = ({ location, yearAndData, className, select }) => {
  let yearAndDataKeys = Object.keys(yearAndData);
  let listItems = yearAndDataKeys.map(key => {
    let img = <img src={checkMark}
                   alt='check mark, above half'/>;
    let liClass = 'aboveHalf';
    if (yearAndData[key] < 0.5) {
      img = <img src={xMark}
                 alt='x mark, below half'/>;
      liClass = 'belowHalf';
    }
    return (<li key={Math.random()}
                className={`hover ${liClass}`}>
              {img}

              {key}: {yearAndData[key]}
            </li>);
  });

  return (

    <div className={`hover card ${className}`}
         onClick={(location) => { select(location); }}>
      <h3 className="hover card-title">{location}</h3>

      <ul className="hover card-list">{listItems}</ul>
    </div>
  );
};

Card.propTypes = {
  location: PropTypes.string.isRequired,
  yearAndData: PropTypes.object.isRequired,
  className: PropTypes.string,
  select: PropTypes.func.isRequired
};

export default Card;
