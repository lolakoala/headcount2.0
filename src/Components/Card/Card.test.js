import React from 'react';
import Card from './Card.js';
import { shallow } from 'enzyme';

describe('Card', () => {
  let yearAndData = { '2001': .99, '2002': .98, '2003': .97 };
  let wrapper = shallow(<Card location='Denver'
                              yearAndData={yearAndData}
                              select={jest.fn()}/>);

  it('should have a district', () => {
    let district = wrapper.find('h3');

    expect(district.text()).toEqual('Denver');
  });

  it('should have an <li> for every key in yearAndData', () => {
    let yearAndDataKeys = Object.keys(yearAndData);
    let listItems = wrapper.find('li');

    expect(listItems.nodes.length).toEqual(yearAndDataKeys.length);
  });

  it('should display key value pairs at <li>', () => {
    let listItems = wrapper.find('li');
    let yearAndDataKeys = Object.keys(yearAndData);

    expect(listItems.nodes[0].props.children.join(''))
    .toEqual(`${yearAndDataKeys[0]}: ${yearAndData[yearAndDataKeys[0]]}`);
    expect(listItems.nodes[1].props.children.join(''))
    .toEqual(`${yearAndDataKeys[1]}: ${yearAndData[yearAndDataKeys[1]]}`);
    expect(listItems.nodes[2].props.children.join(''))
    .toEqual(`${yearAndDataKeys[2]}: ${yearAndData[yearAndDataKeys[2]]}`);
  });

  it('should match the snapShot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
