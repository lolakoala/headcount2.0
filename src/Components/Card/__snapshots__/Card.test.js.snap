// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapShot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="card undefined Denver"
    onClick={[Function]}
>
    <h3
        className="card-title"
    >
        Denver
    </h3>
    <ul>
        <li
            className="aboveHalf"
        >
            2001
            : 
            0.99
        </li>
        <li
            className="aboveHalf"
        >
            2002
            : 
            0.98
        </li>
        <li
            className="aboveHalf"
        >
            2003
            : 
            0.97
        </li>
    </ul>
</div>,
  "nodes": Array [
    <div
      className="card undefined Denver"
      onClick={[Function]}
>
      <h3
            className="card-title"
      >
            Denver
      </h3>
      <ul>
            <li
                  className="aboveHalf"
            >
                  2001
                  : 
                  0.99
            </li>
            <li
                  className="aboveHalf"
            >
                  2002
                  : 
                  0.98
            </li>
            <li
                  className="aboveHalf"
            >
                  2003
                  : 
                  0.97
            </li>
      </ul>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Card
        location="Denver"
        select={[Function]}
        yearAndData={
                Object {
                        "2001": 0.99,
                        "2002": 0.98,
                        "2003": 0.97,
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "location": "Denver",
          "select": [Function],
          "yearAndData": Object {
            "2001": 0.99,
            "2002": 0.98,
            "2003": 0.97,
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="card undefined Denver"
          onClick={[Function]}
>
          <h3
                    className="card-title"
          >
                    Denver
          </h3>
          <ul>
                    <li
                              className="aboveHalf"
                    >
                              2001
                              : 
                              0.99
                    </li>
                    <li
                              className="aboveHalf"
                    >
                              2002
                              : 
                              0.98
                    </li>
                    <li
                              className="aboveHalf"
                    >
                              2003
                              : 
                              0.97
                    </li>
          </ul>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="card undefined Denver"
          onClick={[Function]}
>
          <h3
                    className="card-title"
          >
                    Denver
          </h3>
          <ul>
                    <li
                              className="aboveHalf"
                    >
                              2001
                              : 
                              0.99
                    </li>
                    <li
                              className="aboveHalf"
                    >
                              2002
                              : 
                              0.98
                    </li>
                    <li
                              className="aboveHalf"
                    >
                              2003
                              : 
                              0.97
                    </li>
          </ul>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Card
    location="Denver"
    select={[Function]}
    yearAndData={
        Object {
            "2001": 0.99,
            "2002": 0.98,
            "2003": 0.97,
          }
    }
/>,
}
`;
